ibis/expr/datatypes.py

```diff
@@ -1139,7 +1139,10 @@ def infer_date(value):
 
 @infer.register(datetime.datetime)
 def infer_timestamp(value):
-    return timestamp
+    if value.tzinfo:
+        return Timestamp(timezone=value.tzinfo)
```

Comment:
We don't actually enforce this anywhere, but should we convert `value.tzinfo` to a `str` to be consistent with other uses of the "timestamp with timezone" type in the codebase?